src/viam/sdk/robot/client.hpp

```diff
@@ -23,13 +25,62 @@ namespace viam {
 namespace sdk {
 
 using grpc::Channel;
-using viam::common::v1::ResourceName;
-using viam::common::v1::Transform;
-using viam::robot::v1::FrameSystemConfig;
 using viam::robot::v1::RobotService;
-using viam::robot::v1::Status;
+using time_point = std::chrono::time_point<long long, std::chrono::nanoseconds>;
+
+struct discovery_query {
+    discovery_query(){};
+    viam::robot::v1::DiscoveryQuery to_proto() const;
+    static discovery_query from_proto(const viam::robot::v1::DiscoveryQuery& proto);
+
+    std::string subtype;
+    std::string model;
+    friend bool operator==(const discovery_query& lhs, const discovery_query& rhs);
+};
+
+struct discovery {
+    discovery(){};
+    viam::robot::v1::Discovery to_proto() const;
+    static discovery from_proto(const viam::robot::v1::Discovery& proto);
+
+    discovery_query query;
+    AttributeMap results;
+    friend bool operator==(const discovery& lhs, const discovery& rhs);
+};
+
+struct frameSystemConfig {
+    viam::robot::v1::FrameSystemConfig to_proto() const;
+    static frameSystemConfig from_proto(const viam::robot::v1::FrameSystemConfig& proto);
+    frameSystemConfig(WorldState::transform frame) : frame(std::move(frame)){};
+
+    WorldState::transform frame;
+    AttributeMap kinematics;
+    friend bool operator==(const discovery_query& lhs, const discovery_query& rhs);
+};
+
+struct status {
+    status(){};
+    viam::robot::v1::Status to_proto() const;
+    static status from_proto(const viam::robot::v1::Status& proto);
+
+    boost::optional<Name> name;
+    boost::optional<AttributeMap> status_map;
```

Comment:
`AttributeMap` is a pointer under the hood already, so making this value `optional` on top of that seems a bit redundant and confusing. (What if it exists but is a `nullptr`?) Would suggest making this non-optional (and other optional`AttributeMap` fields below).